pkg/auth/dependency/authenticator/password/provider.go

```diff
@@ -5,9 +5,9 @@ import (
 
 	"github.com/authgear/authgear-server/pkg/auth/config"
 	taskspec "github.com/authgear/authgear-server/pkg/auth/task/spec"
-	"github.com/authgear/authgear-server/pkg/core/errors"
 	"github.com/authgear/authgear-server/pkg/task"
 	"github.com/authgear/authgear-server/pkg/util/clock"
+	"github.com/authgear/authgear-server/pkg/util/errors"
```

Comment:
It would be great if we also remove the overlapping function (with `errors`) in `util/errors`. To avoid renaming when we need to import both, we can rename it to `util/errorutil`.

Reply:
We can remove `errors.New` & `errors.Newf`, and replace them with `errors.New` (from standard `errors` package) and `fmt.Errorf`.

I suppose we can also cleanup other under-utilized functions with little benefits?
- barriers (`errors.HandledWithMessage`)
- secondary errors (`errors.WithSecondaryError`)

Reply:
Keep them please! They are useful. It is just our fault that they are not actively being used.